### PR TITLE
Image overlay

### PR DESCRIPTION
**Pull request by krystalmejia24**

### CODE REVIEW

File: provider/bitmovin/bitmovin.go [hunk 1]
```diff
@@ -742,7 +722,41 @@ func (p *bitmovinProvider) CreatePreset(_ context.Context, preset db.Preset) (st
 		return "", err
 	}
 
-	return svc.Create(preset)
+	presetSummary, err := svc.Create(preset)
```
Comment (krystalmejia24) [inline provider/bitmovin/bitmovin.go:725]: shoud I be tracing the request here? I noticed we weren't before but this does make an api call to bitmovin for a codec configuration.

File: db/types.go [hunk 2]
```diff
@@ -266,7 +271,13 @@ const (
 
 //Overlays defines all the overlay settings for a Video preset
 type Overlays struct {
-	TimecodeBurnin *TimecodeBurnin `json:"timecodeBurnin,omitempty" redish-hash:"timecodeburnin,expand,omitempty"`
+	Images         []Image         `json:"image,omitempty" redis-hash:"image,expand,omitempty"`
```
Comment (zsiec) [inline db/types.go:274]: tiny nit, should the json/redis names be `images` plural?

File: provider/bitmovin/bitmovin.go (hunk 1 above)
Comment (zsiec) [reply provider/bitmovin/bitmovin.go:725]: We want to revisit how we're doing tracing here. I think we can wait until we do that to add it here.

File: provider/bitmovin/bitmovin.go [hunk 3]
```diff
@@ -742,7 +722,41 @@ func (p *bitmovinProvider) CreatePreset(_ context.Context, preset db.Preset) (st
...
+		return "", err
+	}
+
+	if presetSummary.HasVideo() {
+		deInterlace, err := p.api.Encoding.Filters.Deinterlace.Create(model.DeinterlaceFilter{
+			Name:       "deinterlace",
+			AutoEnable: model.DeinterlaceAutoEnable_META_DATA_AND_CONTENT_BASED,
+		})
+		if err != nil {
+			return "", errors.Wrap(err, "creating deinterlace filter")
```
Comment (zsiec) [inline provider/bitmovin/bitmovin.go:736]: We're trying to update our usage of `errors.Wrap`

```suggestion
			return "", fmt.Errorf("creating deinterlace filter: %w", err)

File: provider/bitmovin/bitmovin.go [hunk 4]
```diff
@@ -742,7 +722,41 @@ func (p *bitmovinProvider) CreatePreset(_ context.Context, preset db.Preset) (st
...
+		}
+
+		presetSummary.VideoFilters = append(presetSummary.VideoFilters, deInterlace.Id)
+
+		if overlays := preset.Video.Overlays; overlays != nil && overlays.Images != nil {
```
Comment (zsiec) [inline provider/bitmovin/bitmovin.go:741]: Depending on how many images we end up supporting, we may want to eventually create these in parallel.

File: provider/bitmovin/bitmovin.go [hunk 5]
```diff
@@ -742,7 +722,41 @@ func (p *bitmovinProvider) CreatePreset(_ context.Context, preset db.Preset) (st
...
+			for _, image := range overlays.Images {
+				watermark, err := p.api.Encoding.Filters.Watermark.Create(model.WatermarkFilter{
```
Comment (zsiec) [inline provider/bitmovin/bitmovin.go:743]: We may want to eventually consider creating these in parallel.

File: db/types.go (hunk 2 above)
Comment (krystalmejia24) [reply db/types.go:274]: good catch! i didn't update the tags in the client either.

**Pull request merged**